src/SAWScript/Crucible/LLVM/X86.hs

```diff
@@ -623,7 +641,38 @@ assertPost env premem preregs = do
   liftIO $ C.addAssertion sym . C.LabeledPred correctStack . C.SimError W4.initializationLoc
     $ C.AssertFailureSimError "Stack not preserved" ""
 
-  -- TODO: Match return value in RAX
+  pointsToMatches <- forM (ms ^. MS.csPostState . MS.csPointsTos)
+    $ assertPointsTo env tyenv nameEnv
+
+  returnMatches <- case (ms ^. MS.csRetValue, ms ^. MS.csRet) of
+    (Just expectedRet, Just retTy) -> do
+      postRAX <- C.LLVM.ptrToPtrVal <$> getReg Macaw.RAX postregs
+      pure [LO.matchArg opts sc cc ms MS.PostState postRAX retTy expectedRet]
+    _ -> pure []
+
+  let setupConditionMatches = LO.learnSetupCondition opts sc cc ms MS.PostState
+        <$> (ms ^. MS.csPostState . MS.csConditions)
```

Comment:
I prefer an explicit `map` instead of `<$>` from the list monad. It's just harder to read.